src/beliefpropagation/beliefpropagation.jl

```diff
@@ -169,48 +135,46 @@ function belief_propagation(
 end
 
 function belief_propagation(
-  tn::ITensorNetwork;
-  contract_kwargs=(; alg="density_matrix", output_structure=path_graph_structure, maxdim=1),
-  nvertices_per_partition=nothing,
-  npartitions=nothing,
-  subgraph_vertices=nothing,
-  niters=default_bp_niters(mts),
-  target_precision=nothing,
-  verbose=false,
+  ptn::PartitionedGraph;
+  itensor_constructor=inds_e -> ITensor[dense(delta(inds_e))],
+  kwargs...,
 )
-  mts = message_tensors(tn; nvertices_per_partition, npartitions, subgraph_vertices)
-  return belief_propagation(tn, mts; contract_kwargs, niters, target_precision, verbose)
+  mts = message_tensors(ptn; itensor_constructor)
+  return belief_propagation(ptn, mts; kwargs...)
 end
-
 """
 Given a subet of vertices of a given Tensor Network and the Message Tensors for that network, return a Dictionary with the involved subgraphs as keys and the vector of tensors associated with that subgraph as values
 Specifically, the contraction of the environment tensors and tn[vertices] will be a scalar.
 """
-function get_environment(tn::ITensorNetwork, mts::DataGraph, verts::Vector; dir=:in)
-  subgraphs = unique([find_subgraph(v, mts) for v in verts])
+function get_environment(ptn::PartitionedGraph, mts, verts::Vector; dir=:in)
+  partition_verts = partitionvertices(ptn, verts)
+  central_verts = vertices(ptn, partition_verts)
 
   if dir == :out
-    return get_environment(tn, mts, setdiff(vertices(tn), verts))
+    return get_environment(ptn, mts, setdiff(vertices(ptn), verts))
   end
 
-  env_tns = ITensorNetwork[mts[e] for e in boundary_edges(mts, subgraphs; dir=:in)]
-  central_tn = ITensorNetwork(
-    ITensor[tn[v] for v in setdiff(flatten([vertices(mts[s]) for s in subgraphs]), verts)]
-  )
-  return ITensorNetwork(vcat(env_tns, ITensorNetwork[central_tn]))
+  pedges = partitionedges(ptn, boundary_edges(ptn, central_verts; dir=:in))
+  env_tensors = [mts[e] for e in pedges]
+  env_tensors = reduce(vcat, env_tensors; init=ITensor[])
+  central_tensors = ITensor[
+    (unpartitioned_graph(ptn))[v] for v in setdiff(central_verts, verts)
+  ]
+
+  return vcat(env_tensors, central_tensors)
 end
 
 """
 Calculate the contraction of a tensor network centred on the vertices verts. Using message tensors.
 Defaults to using tn[verts] as the local network but can be overriden
 """
 function approx_network_region(
```

Comment:
Like with `get_environment` it seems like this should have versions that accept partition vertices or underlying graph vertices.

Reply:
Agreed. Let me write those out.

Reply:
Also, `approx_network_region` seems like a strange function to me. Maybe it isn't really needed at this point given the added convenience of `PartitionedGraph`, since it really is just grabbing the environment and then tensors in a partition?

Reply:
Yeah it is just grabbing the environment and then concatenating the `ITensors` with some local tensors on the partition. I think removing it would be fine actually.